src/epics/timeline.js

```diff
@@ -0,0 +1,66 @@
+import { Observable } from 'rxjs';
+import { Map, OrderedMap } from 'immutable';
+import * as TimelineActions from 'app/actions/timeline';
+import RevisionEntity from 'app/entities/revision';
+
+const getWikiDomain = ( wiki, wikis ) => {
+	let wikiRecord = wikis.find( ( obj ) => {
+		return obj.get( 'id' ) === wiki;
+	} );
+
+	return wikiRecord.domain;
+};
+
+const buildUrl = ( store ) => {
+	const domain = getWikiDomain( store.getState().query.wiki, store.getState().wikis );
+	const users = store.getState().query.user.join( '|' );
+	const ucstart = store.getState().query.startDate;
+
+	let url = 'https://' + domain + '/w/api.php?action=query&list=usercontribs&ucuser=' + users + '&ucstart=' + ucstart + '&uclimit=500&ucdir=older&format=json&origin=*';
+
+	return url;
+};
+
+const fetchTimeline = ( action$, store ) => (
+	action$.filter( ( action ) => {
+		return [ 'QUERY_UPDATE', 'QUERY_SET_VALUE', 'WIKIS_SET' ].includes( action.type );
+	} )
+		.filter( () => {
+			return !!store.getState().wikis.size;
```

Comment:
I would replace this with something like this:
https://github.com/wikimedia/takedown-tools/blob/9d34e0663e087a433e15378adfb6d15d388178f8/client/epics/site.js#L70-L71
and then you can remove `WIKI_SET` from the list above.